test/integration/imagechange_buildtrigger_test.go

```diff
@@ -477,12 +479,21 @@ func TestMultipleImageChangeBuildTriggers(t *testing.T) {
 		if err != nil {
 			t.Fatalf("Couldn't create Image: %v", err)
 		}
+
+		var newBuild *buildapi.Build
+		var event watchapi.Event
 		// wait for initial build event from the creation of the imagerepo
-		event := <-watch.ResultChan()
+		for {
+			event = <-watch.ResultChan()
+			newBuild = event.Object.(*buildapi.Build)
+			if _, exists := ignoreBuilds[newBuild.Name]; !exists {
```

Comment:
no bike shedding intended, but wdyt of 

```go
if _, seen := buildsSeen[build.Name]; !seen {
```

It took me some time to decode what are "ignoreBuilds" and how they fit in the algorithm.
IIUC they are "ignored" because they've been already seen?

Also worth dropping a comment why we need to do this filtering for whoever is reading this test case in the future.